lib/mongo/uri.rb

```diff
@@ -12,7 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-require 'uri'
+require 'addressable'
```

Comment:
Can you explain this please? Why should we create a dependency on another gem instead of using a core Ruby module?